File: Python/Recursion-1/count7.py

# given a non-negative int n, return the count of the occurrences of 7 as a digit, so for example 717 yields 2
def count7(n):
    count = 0

    if int(n) == 0:
        return count
    if int(n % 10) == 7:
        count += 1
    
    return count + count7(n/10)

print(count7(717))
print(count7(7))
print(count7(1))

File: Java/Recursion-1/count7.java

public class recursion1 {
	
	// given a non-negative int n, return the count of the occurrences of 7 as a digit, so for example 717 yields 2
	public int count7(int n) {
		
		int count = 0;
		
		if (n == 0) {
			return count;
		}

		if (n % 10 == 7) {
			count++;
		}
		
		return count + count7(n/10);
		
	}

	public static void main(String[] args) {

		recursion1 o = new recursion1();
		
		System.out.println(o.count7(717));
		System.out.println(o.count7(7));
		System.out.println(o.count7(123));

	}

}
